### PR TITLE
Ability to trigger a jenkins job from octo

### PR DESCRIPTION
**Pull request by george-nelson**

### CODE REVIEW

File: step-templates/Jenkins-Queue-Job [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+    "Octopus.Action.Script.ScriptFileName": null,
+    "Octopus.Action.Package.FeedId": null,
+    "Octopus.Action.Package.PackageId": null
+  },
+  "Parameters": [
+    {
+      "Id": "b8337514-3989-4b33-930c-b5ebde5b4be0",
+      "Name": "JobUrl",
```
Comment (alfhenrik) [inline step-templates/Jenkins-Queue-Job]: Could you please prefix each of the parameters with `qjj_` (and update the script body with the new variable names) to minimize the risk of duplicate variables in a project.

File: step-templates/Jenkins-Queue-Job [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+        "Octopus.ControlType": "SingleLineText"
+      },
+      "Links": {}
+    },
+    {
+      "Id": "a52f7318-6f45-4e9f-b825-b3ae767608f8",
+      "Name": "FailBuild",
+      "Label": "Fail Build",
+      "HelpText": "Should this fail the deployment?",
+      "DefaultValue": "No",
```
Comment (alfhenrik) [inline step-templates/Jenkins-Queue-Job]: This needs to be `false` to work

File: step-templates/Jenkins-Queue-Job [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
...s['JobUrl']\r\n$failBuild = $OctopusParameters['FailBuild']\r\n$jobTimeout = $OctopusParameters['JobTimeout']\r\n$buildParam = $OctopusParameters['BuildParam']\r\n$buildResult = \"/lastBuild/api/json?tree=result,number,building\"\r\n\r\nWrite-Host \"job url: \" $jobURL$buildParam \r\n$response = Invoke-WebRequest -Uri $jobURL$buildParam  -Method Post\r\n\r\nif ($failBuild -eq \"True\")\r\n{\r\n   $isBuilding=\"True\"\r\n   $i = 0\r\n   Write-Host \"Estimate Job Duration: \" $jobTimeout\r\n   while ($isBuilding -eq \"True\" -and $i -lt $jobTimeout)\r\n   {       \r\n       $i +=20\r\n       Write-Host \"waiting $i secs for build to complete\"\r\n       Start-Sleep -s 20\r\n       $retyJobStatus = Invoke-RestMethod -Uri $jobURL$buildResult -Method Get\r\n       $isBuilding=$retyJobStatus[0].building\r\n       $result = $retyJobStatus[0].result\r\n       $buildNumber = $retyJobStatus[0].number\r\n       Write-Host \"Retry Job Status: \" $result \" BuildNumber: \"  $buildNumber  \" IsBuilding: \"  $isBuilding \r\n   }\r\n   \r\n   if ($result -ne \"SUCCESS\")\r\n   {\r\n       echo \"BUILD FAILURE: build is unsuccessful or status could not be obtained.\"\r\n       exit 1\r\n   }\r\n}",
```
Comment (alfhenrik) [inline step-templates/Jenkins-Queue-Job]: `$JenkinsServer = "http://jenkins:8080/` probably needs to be a configurable value as this seems very particular to your setup

Comment (alfhenrik) [inline step-templates/Jenkins-Queue-Job]:
```
$i+=20
Write-Host "waiting $i secs for build to complete"
Start-Sleep -s 20
```
Either the message should say `waiting 20 secs...` or `Start-Sleep` should use the `$i` variable

Comment (alfhenrik) [inline step-templates/Jenkins-Queue-Job]: `$retyJobStatus = Invoke-RestMethod -Uri $jobURL$buildResult -Method Get`
Should be `$retryJobStatus = ...`

File: step-templates/Jenkins-Queue-Job [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+      "HelpText": "e.g. ?Param=Value or ?delay=10sec",
+      "DefaultValue": "/build",
+      "DisplaySettings": {
+        "Octopus.ControlType": "SingleLineText"
+      },
+      "Links": {}
+    }
+  ],
+  "LastModifiedOn": 2017-08-11T15:50:26.518Z",
+  "LastModifiedBy": "George Nelson",
```
Comment (alfhenrik) [inline step-templates/Jenkins-Queue-Job]: If you update this to your :octocat: handle, it will be nicely linked on the Library website

File: step-templates/Jenkins-Queue-Job [hunk 5]
```diff
@@ -0,0 +1,71 @@
+{
+  "Id": "ccb7ad4c-a19e-426f-822e-cd0e0243bda3",
+  "Name": "Queue Jenkins Job",
```
Comment (alfhenrik) [inline step-templates/Jenkins-Queue-Job]: Could you reword this to `Jenkins - Queue Job` please

File: step-templates/Jenkins-Queue-Job [hunk 6]
```diff
@@ -0,0 +1,71 @@
+{
```
Comment (alfhenrik) [inline step-templates/Jenkins-Queue-Job]: Filename must include the `.json` extension

**Comment by george-nelson** [comment]
Sorry for my late notice. I just noticed your feedback. Will get this addressed soon.

**Comment by george-nelson** [comment]
Awesome thank you

Sent from my iPhone

On Sep 24, 2017, at 6:25 PM, Henrik Andersson <notifications@github.com<mailto:notifications@github.com>> wrote:


[https://camo.githubusercontent.com/31231ddfcd8bcbf948638bcf21528cc89eeeba05/68747470733a2f2f6d656469612e67697068792e636f6d2f6d656469612f4f4536464534475a4637386e6d2f67697068792e676966]<https://camo.githubusercontent.com/31231ddfcd8bcbf948638bcf21528cc89eeeba05/68747470733a2f2f6d656469612e67697068792e636f6d2f6d656469612f4f4536464534475a4637386e6d2f67697068792e676966>
